core/src/test/java/org/elasticsearch/index/mapper/CompletionFieldMapperTests.java

```diff
@@ -163,6 +163,28 @@ public void testParsingMinimal() throws Exception {
         assertSuggestFields(fields, 1);
     }
 
+    public void testParsingFailure() throws Exception {
+        String mapping = jsonBuilder().startObject().startObject("type1")
+            .startObject("properties").startObject("completion")
+            .field("type", "completion")
+            .endObject().endObject()
+            .endObject().endObject().string();
+
+        DocumentMapper defaultMapper = createIndex("test").mapperService().documentMapperParser().parse("type1", new CompressedXContent(mapping));
+
+        try {
+            defaultMapper.parse(SourceToParse.source("test", "type1", "1", XContentFactory.jsonBuilder()
```

Comment:
you could use expectThrows here

Reply:
Ah thanks, I was just about to ask for a better way to do this :)